src/Quadrature/quadrature.jl

```diff
@@ -40,6 +40,11 @@ struct QuadratureRule{dim,shape,T}
     points::Vector{Vec{dim,T}}
 end
 
+function QuadratureRule{shape}(weights::AbstractVector{Tw}, points::AbstractVector{Vec{dim,Tp}}) where {dim, shape, Tw, Tp}
```

Comment:
I think it makes sense to require `dim` here (even though it can be inferred from `points`) since otherwise the first "parameter" of `QuadratureRule` in this constructor does not match the first parameter of the returned value:
```suggestion
function QuadratureRule{dim, shape}(weights::AbstractVector{Tw}, points::AbstractVector{Vec{dim,Tp}}) where {dim, shape, Tw, Tp}
```